akka-actor-tests/src/test/scala/akka/pattern/BackoffSupervisorSpec.scala

```diff
@@ -203,11 +202,36 @@ class BackoffSupervisorSpec extends AkkaSpec with ImplicitSender with Eventually
       }
     }
 
+    "reply to sender if withHandlerWhileStopped is specified" in {
+      filterException[TestException] {
+        val supervisor = create(
+          BackoffOpts
+              .onFailure(Child.props(testActor), "c1", 100.seconds, 300.seconds, 0.2)
+              .withMaxNrOfRetries(-1)
+              .withHandlerWhileStopped({case (msg, sender) => sender ! TestReply(msg)}))
```

Comment:
While I understand that this gives the ultimate flexibility, it looks a bit dangerous to me because easy to accidentally/unsafely close over the surrounding scope, it essentially is giving an actor (the supervisor) a receive block from the outside.

Could it perhaps be a stand in actor props instead?

Reply:
You mean props of a custom actor receiving the message and replying while the supervised is stopped?

Reply:
Yep, not sure it is the greatest of ideas but it would be safer.

Reply:
I like the idea, it does not open the API for easy to make mistakes while providing the flexibility we need :)

Reply:
I've pushed a new commit with the proposed change